x-pack/docs/en/rest-api/security/update-user-profile-data.asciidoc

```diff
@@ -1,10 +1,173 @@
+being referenced in the request
+
+[[security-api-update-user-profile-data-desc]]
+==== {api-description-title}
+
+This API allows update to both the `access` and `data` fields of an existing
+user profile document with JSON objects.
+Data provided in the request are merged into existing data:
+
+* New keys and their values are added to document.
+* Conflicting keys will be replaced by the data provided in the request.
+
+For both `access` and `data`, the data are namespaced by the top level fields.
+The _"update profile data"_ global privilege allows granting privilege for
+updating only the allowed namespaces.
+
+[[security-api-update-user-profile-data-path-params]]
+==== {api-path-parms-title}
+
+`uid`::
+(Required, string) An unique identifier for the user profile.
+
+[[security-api-update-user-profile-data-query-params]]
+==== {api-query-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=if_seq_no]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=if_primary_term]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=refresh]
+
+`uid`::
+(Required, string) An unique identifier for the user profile.
+
+[[security-api-update-user-profile-data-request-body]]
+==== {api-request-body-title}
+
+The following parameters can be specified in the body of a POST request:
+
+`access`::
+(Required*, object)
+Arbitrary searchable data that you want to associate with the user profile.
+It supports nested data structure. Within the `access` object, top level
+keys cannot begin with `_` or contain `.`.
+
+`data`::
+(Required*, object)
+Arbitrary non-searchable data that you want to associate with the user profile.
+It supports nested data structure. Within the `data` object, top level
+keys cannot begin with `_` or contain `.`.
+The `data` object is not searchable, but can be retrieved with <<security-api-get-user-profile>>.
+
+--
+*Indicates that the setting is required in some, but not all situations.
+--
+
+[[security-api-update-user-profile-data-example]]
+==== {api-examples-title}
+
+To update the profile data, submit a POST request to the `/_security/profile/_data/<uid>`
+endpoint. Assuming a profile document exist with uid `u_kd2JMqwUQwSCCOxMv7M1vw`,
+its data can be updated with the following:
+
+[source,console]
+------------------------------------------------------------
+POST /_security/profile/_data/u_kd2JMqwUQwSCCOxMv7M1vw
+{
+  "access": {
+    "app1": {
+      "tag": "prod"
+    }
+  },
+  "data": {
+    "app1": {
+      "theme": "default"
+    }
+  }
+}
+------------------------------------------------------------
+// TEST[skip:uid is random and no way to ensure this uid exists]
+
+A successful update profile data API call returns a JSON structure that shows
+the request is acknowledged.
+
+[source,js]
+--------------------------------------------------
+{
+  "acknowledged": true
+}
+--------------------------------------------------
+// NOTCONSOLE
```

Comment:
I think that this information should be in a Response body section.

`==== {api-response-body-title}`

Reply:
Relocated.